docs/admin/authentication.md

```diff
@@ -486,6 +486,21 @@ An unsuccessful request would return:
 
 HTTP status codes can be used to supply additional error context.
 
+#### Using OpenStack Keystone authentication Webhook
+
+There are two parts to authentication with OpenStack Keystone. On the client-side, you can use `--auth-provider=openstack` option to reuse the `OS_*` environment variables:
+```
+kubectl config set-credentials openstackuser --auth-provider=openstack
+```
+Note that you will also need to use `kubectl config set-context` and `kubectl config use-context` to activate the context. Please make sure you have `OS_DOMAIN_NAME` to use Keystone V3 API.
+
+On the server-side, one choice is to use the webhook from:   
+https://github.com/dims/openstack-cloud-controller-manager/blob/master/cmd/k8s-keystone-auth/main.go
```

Comment:
I don't know that referencing the Go code will help an administrator know how to use this, but if you want to reference it, I recommend something like:

>  The OpenStack Cloud Controller includes an [webhook authenticator](https://github.com/dims/openstack-cloud-controller-manager/blob/master/cmd/k8s-keystone-auth/main.go).